### PR TITLE
Setup/develop file architecture

### LINKED ISSUE #3: Component Architecture
- [ ] Develop component architecture
- [ ] Forecast functional/ class components
- [ ] Plan flow of data

### PR DESCRIPTION
**Pull request by rachellhill**
# Description
Please include a summary of the changes and/or which issue is fixed. Please also include relevant motivation and context. 

- Created initial file structure 
- Successfully fetched from brewery API 
- Implemented TypeScript in App file

Were there any issues that arose? 

- Trouble running typescript

Is there anything you need from your teammates? 

Any other comments, questions, or concerns?

## Issues
Closes #3 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Cosmetic changes (changes to layout or visuals in CSS file)

# How Has This Been Tested?
- [x] React Dev Tools
- [x] Functionality check through web browser
- [ ] Ran Cypress test suite

# Checklist:
- [x] My code follows the style guidelines of this project (semicolons, indentation, naming conventions, etc..).
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have deleted all comments after functionality and understanding is final.
- [x] I have added any issues closed by this PR in the Issues section.
- [x] I have added others as reviewers.

### CODE REVIEW

File: src/Components/App.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+// state is going to have one key which is breweries
+  // breweries is an array that contains objects which are the interface of brewery
+type State = {
+  breweries: Brewery[]
+}
+
+// Brewery is an object that has all of these keys, which have specified data types as their values
+  // each element in our breweries state is one of these objects
+interface Brewery {
```
Comment (userigorgithub) [inline src/Components/App.tsx:12]: Great job making types

File: src/Components/App.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  longitude: string,
+  latitude: string,
+  phone: string,
+  website_url: string,
+  updated_at: string,
+  created_at: string
+}
+
+// when setting up a class component using typescript, pass an empty object if no props are being passed down
+  // expecting two arguments: props and state
```
Comment (userigorgithub) [inline src/Components/App.tsx:33]: Good job creating TS class

File: src/Components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  postal_code: string,
+  country: string,
+  longitude: string,
+  latitude: string,
+  phone: string,
+  website_url: string,
+  updated_at: string,
+  created_at: string
+}
+
```
Comment (stephanie-roe) [inline src/Components/App.tsx:31]: This is awesome! I think later on we can refactor this into another file.

**Review by stephanie-roe — APPROVED** [review]
This is looking great! I'm so happy we were able to navigate the newness of TS and get our API calls up and running.

File: src/Components/App.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  state: State = {
+    breweries: [],
+  }
+
+// invoking our fetch and then setting the state to include all of the brewery data we just received from our promise
+  componentDidMount() {
+    this.getAllBreweries().then(breweries => this.setState(state => ({ breweries: breweries })))
+  }
+
+// this method is given a type (<Brewery[]> --> going to be an array and everything in that array will be this Brewery object). the type communicates our anticipated outcome. We are expecting a promise which is an array that contains Brewery objects
```
Comment (userigorgithub) [inline src/Components/App.tsx:44]: Awesome! Fetching data using TS!

**Review by userigorgithub — APPROVED** [review]
Great start with using TS in App.tsx file.

**Pull request merged**